chore(issues-search): make issue type searches more human readable

MichaelSun48 opened the pull request:
Described in #45195 

When using the `issues:type` tag in search, each option in the dropdown menu will include a section to the right that is more human readable.
Previous:                              |  New:
:-------------------------:|:-------------------------:
<img width="707" alt="image" src="https://github.com/getsentry/sentry/assets/55160142/e97062e2-1b20-4a5b-9b37-057913174b34"> | <img width="633" alt="image" src="https://github.com/getsentry/sentry/assets/55160142/efcf9b95-dbe2-4645-bf80-c03c4a1a040f">


Extras:
- Updated `IssueTitle` enum in  `group.tsx` to include all issue types. Descriptions were copy pasted from the backend (`grouptype.py`), which included previously absent issue titles.

Commit:
Added human-friendly description section under Issues search, issue:type

Commit:
Updated IssueTitles according to backend descriptions (grouptype.py), added mapping from issue type to issue title

MichaelSun48 committed:
Merge branch 'master' into michaelsun/issuesSearchDropdownImprovements_45195

MichaelSun48 committed:
Merge branch 'master' into michaelsun/issuesSearchDropdownImprovements_45195

john-z-yang committed:
Fix(metrics): Fix indexer telemetry (#65753)

### Overview

Fix calculation of `max_value_len`

scttcper commented on static/app/types/group.tsx:
```diff
@@ -110,6 +111,54 @@ export enum IssueTitle {
+  REPLAY_RAGE_CLICK = 'Rage Click Detected',
+}
+
+const ISSUE_TYPE_TO_ISSUE_TITLE = {
```
is this the same thing as the enum above but in lower case?

MichaelSun48 replied:
The keys for this dict? Yes they are. I couldn't find a non-janky way to match an IssueType value (like `performance_db_main_thread`) back to it's key (`PERFORMANCE_DB_MAIN_THREAD`) so we could get the IssueTitle enum with the same key (`DB on Main Thread`). Do you know if there's a better way to do this other than creating a separate (and kinda redundant) dict?

scttcper replied:
Thanks that clears up my question 👍 added a different suggestion

narsaynorath committed:
feat(app-start): Add perf callout for onboarding (#65789)

roggenkemper commented on static/app/components/smartSearchBar/searchDropdown.tsx:
```diff
@@ -372,6 +372,23 @@ function DropdownItem({
         <RecommendedItemTitle>{item.title}</RecommendedItemTitle>
       </RecommendedItem>
     );
+  } else if (item.type === ItemType.TAG_VALUE) {
```
this seems to be mostly copied over from the else with a couple lines changes. could we just add the `documentation` field to `item` for these item types instead of having a new else if

snigdhas committed:
fix(issue-priority): Only update priority and activity if priority changes (#65751)

Checks the current priority when updating, to only write to
GroupHistory/Activity tables if the priority is changed.

Fixes https://github.com/getsentry/sentry/issues/65663

scttcper committed:
deps(ui): Upgrade framer-motion to v6.5.1 (#65742)

cathteng committed:
chore(team-org-roles): remove Organization "get_teams_with_org_role" (#65624)

scttcper committed:
chore(ui): Add RouterProps children for react 18 (#65759)

scttcper committed:
fix(ui): Declare context type on more class components (#65760)

scttcper commented on static/app/types/group.tsx:
```diff
@@ -110,6 +111,54 @@ export enum IssueTitle {
+  replay_click_rage: 'Rage Click Detected',
+};
+
+export function getIssueTitleFromType(issueType: string): IssueTitle | undefined {
+  if (issueType in ISSUE_TYPE_TO_ISSUE_TITLE) {
+    return ISSUE_TYPE_TO_ISSUE_TITLE[issueType];
+  }
+  return undefined;
 }
```
maybe this?
```suggestion
export function getIssueTitleFromType(issueType: string): IssueTitle | undefined {
  if (issueType.toLowerCase() in IssueTitle) {
    return IssueTitle[issueType.toLowerCase()];
  }
  return undefined;
}
```

MichaelSun48 replied:
Some of the issueType values are slightly different from their keys, (e.g. `PERFORMANCE_RENDER_BLOCKING_ASSET = 'performance_render_blocking_asset_span',`), so we'd have to change the enum keys/values to make this work, and I'm not sure what the blast radius of that would be which makes me nervous. Thoughts?

markstory committed:
fix(hybridcloud) Improve webhook delivery logs (#65790)

Add logging where it was missing and add shorten log message names

scttcper committed:
fix(ui): Change a few types for react 18 (#65762)

joshuarli committed:
fix(devenv): no sync in direnv (#65799)

so in `22cac362062` i assumed that `if grep -Eq "^devenv($|>)"
"${SENTRY_ROOT}/requirements-dev.txt";` was working so i moved `devenv
sync` to the happy path, surprise it never worked

people do not like `devenv sync` on cd as it's just too slow and i agree

markstory committed:
fix(hybridcloud) Fix silo limit error in github:e (#65788)

Fixes HC-1124
Fixes SENTRY-2KCC

cmanallen committed:
fix(replays): Add custom replay paginator (#65724)

Adds a custom replay paginator to handle cases where the search and sort
query returns more results than the data query. We want to ensure that
the next page is always fetch-able _if it has the possibility of
existing_.

Pagination works on the following logic:

- Fetch the `limit` + 1.
- If `limit` + 1 items found then the resource has a next page.
- Pop the last item and return the items to the client with a next page
header.
- If < `limit` + 1 items found then the resource has no next page.
    - Return the items to the client without a next page header.

Replays makes two queries. A search and sort query and a data query. The
data query has an assumption baked in that the *scalar* search and sort
query does not. There must be a segment 0. The scalar query does not
enforce this constraint because its trying to find at least one varying
field and that varying field might not be on segment 0.

The search and sort query will return `limit` + 1 items. But because any
of those replays may be missing their segment 0 entry they will be
removed from the data query. The paginator sees `< limit + 1` and
assumes there must not be a second page.

We fix this by returning the next page header if the search and sort
query thinks there's a next page -- not the data query.

michellewzhang committed:
feat(feedback): adjust copy wording and add another more platforms (#65797)

relates to https://github.com/getsentry/sentry/issues/65728

mrduncan committed:
feat(hc): Add new hybrid_cloud.region-domain-allow-list option (#65756)

See getsentry/getsentry#13057
